doc/arch/adr-011-sagemaker.md

```diff
@@ -0,0 +1,141 @@
+|    |                 |       |                      |    |
+|    |    Search API   +------->   Docker container   |    |
+|    |                 <-------+ (TensorFlow Serving) |    |
+|    |                 |       |                      |    |
+|    +-----------------+       +----------------------+    |
+|                                                          |
+|                                                          |
++----------------------------------------------------------+
+```
+
+## After Sagemaker
+
+After introducing Sagemaker, our infrastructure for training
+and serving models does not use Docker and all training and
+serving is done in AWS.
+
+- Both models training and serving is done by AWS Sagemaker
+  instances
+- Search API reranks search results by making requests to a
+  Sagemaker endpoint
+- To create new models, we trigger changes to Sagemaker from
+  a [Concourse][] pipeline.
+- New models are trained in AWS Sagemaker spot instances.
+
+```
+                               +------------------------------+
+    +----------------+         | AWS Sagemaker                |
+    | External model |         |                              |
+    | build trigger  +----+    |   +----------------------+   |
+    |  (Concourse)   |    |    |   |                      |   |
+    +----------------+    +-------->  Sagemaker training  |   |
+                               |   |                      |   |
++---------------------------+  |   +---------+------------+   |
+| Search machine            |  |             |                |
+|                           |  |             |                |
+|    +------------------+   |  |   +---------v----------+     |
+|    |                  |   |  |   |                    |     |
+|    |    Search API    +----------> Sagemaker endpoint |     |
+|    |                  <----------+    (TF serving)    |     |
+|    |                  |   |  |   |                    |     |
+|    +------------------+   |  |   +--------------------+     |
+|                           |  |                              |
+|                           |  |                              |
++---------------------------+  +------------------------------+
+```
+
+From Search API's perspective, nothing has changed. It still
+makes a call to an endpoint which then re-ranks results.
+
+However, externally, Docker is no longer running on search machines,
+and the reranking is performed on a different machine.
+
+Rather than orchestrating model changes on search machines,
+a Concourse pipeline handles the process of creating, training,
+and serving models.
+
+## Downsides
+
+This new architecture has some downsides:
+
+- the search architecture is more complicated, involving more machines
+- we now have a bigger AWS bill, since we have to pay for Sagemaker
+instances
+- Search API requests to Sagemaker instances are considerably slower
+than request to local docker containers, and less reliable
```

Comment:
Go for "potentially less reliable" as I don't think we've actually seen reliability issues, have we?